recipes/google-cloud-cpp-core/meta.yaml

```
@@ -0,0 +1,387 @@
+        # https://github.com/googleapis/google-cloud-cpp/blob/main/doc/adr/2023-05-03-patch-releases-are-ABI-compatible.md
+        - {{ pin_subpackage("libgoogle-cloud", max_pin="x.x") }}
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - libgrpc
+        - libprotobuf
+      host:
+        - libabseil
+        - libcurl
+        - libgrpc
+        - libprotobuf
+        - openssl
+        - nlohmann_json
+        - libcxx  # [osx]
+      run:
+        - libcxx  # [osx]
+      run_constrained:
+        - libgoogle-cloud {{ version }} *_{{ PKG_BUILDNUM }}
+    test:
+      commands:
+        # presence of shared library (unix)
+        - test -f $PREFIX/lib/libgoogle_cloud_cpp_common.{{ version }}.dylib  # [osx]
+        - test -f $PREFIX/lib/libgoogle_cloud_cpp_common.so.{{ version }}  # [linux]
+        # abscense of static library (windows). It belongs only in devel package.
+        - if exist %LIBRARY_LIB%\google_cloud_cpp_common.lib exit 1  # [win]
+
+        # absence of headers (they belong in devel package)
+        - test ! -f $PREFIX/include/google/cloud/version.h  # [not win]
+        - if exist %LIBRARY_INC%\google\cloud\version.h exit 1  # [win]
+
+        # absence of metadata for CMake & pkgconfig (belongs in devel package)
+        - test ! -f $PREFIX/lib/pkgconfig/google_cloud_cpp_common.pc  # [not win]
+        - test ! -f $PREFIX/lib/cmake/google_cloud_cpp_common/google_cloud_cpp_common-config.cmake  # [not win]
+        - if exist %LIBRARY_LIB%\cmake\google_cloud_cpp_common\google_cloud_cpp_common-config.cmake exit 1  # [win]
+
+  - name: libgoogle-cloud-devel
+    script: install-libgoogle-cloud.sh   # [not win]
+    script: install-libgoogle-cloud.bat  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage("libgoogle-cloud", max_pin="x.x") }}
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - libcurl
+        - libgrpc
+        - libprotobuf
+        - openssl
+        - nlohmann_json
+        # cannot load package without zlib because it's specified
+        # as required; if possible, this should be removed upstream
+        - zlib
+      host:
+        - {{ pin_subpackage("libgoogle-cloud", exact=True) }}
+        - libabseil
+        - libcurl
+        - libgrpc
+        - libprotobuf
+        - openssl
+        - nlohmann_json
+      run:
+        - {{ pin_subpackage("libgoogle-cloud", exact=True) }}
+        - libabseil
+        - libcurl
+        - libgrpc
+        - libprotobuf
+        - openssl
+        - nlohmann_json
+        # cannot load package without zlib because it's specified
+        # as required; if possible, this should be removed upstream
+        - zlib        
+        - libcxx  # [osx]
+    test:
+      commands:
+        - ./run_test_common.sh   # [not win]
+        - ./run_test_common.bat  # [win]
```

We can add the header test here. Also a nit to use `# [unix]` instead of `# [not win]` -- more accurate and easier to follow without the negation.
```suggestion
      commands:
        # presence of headers
        - test -f $PREFIX/include/google/cloud/version.h            # [unix]
        - if not exist %LIBRARY_INC%\google\cloud\version.h exit 1  # [win]
        # build an example
        - ./run_test_common.sh   # [unix]
        - ./run_test_common.bat  # [win]
```

---

Done.  Though building the example implicitly tests for the presence of the same header.